airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/AzureBlobStorageConsumer.java

```diff
@@ -0,0 +1,186 @@
+  private final AzureBlobStorageWriterFactory writerFactory;
+  private final Consumer<AirbyteMessage> outputRecordCollector;
+  private final Map<AirbyteStreamNameNamespacePair, AzureBlobStorageWriter> streamNameAndNamespaceToWriters;
+
+  private AirbyteMessage lastStateMessage = null;
+
+  public AzureBlobStorageConsumer(
+                                  AzureBlobStorageDestinationConfig azureBlobStorageDestinationConfig,
+                                  ConfiguredAirbyteCatalog configuredCatalog,
+                                  AzureBlobStorageWriterFactory writerFactory,
+                                  Consumer<AirbyteMessage> outputRecordCollector) {
+    this.azureBlobStorageDestinationConfig = azureBlobStorageDestinationConfig;
+    this.configuredCatalog = configuredCatalog;
+    this.writerFactory = writerFactory;
+    this.outputRecordCollector = outputRecordCollector;
+    this.streamNameAndNamespaceToWriters = new HashMap<>(configuredCatalog.getStreams().size());
+  }
+
+  @Override
+  protected void startTracked() throws Exception {
+    // Init the client itself here
+    StorageSharedKeyCredential credential = new StorageSharedKeyCredential(
+        azureBlobStorageDestinationConfig.getAccountName(),
+        azureBlobStorageDestinationConfig.getAccountKey());
+
+    SpecializedBlobClientBuilder specializedBlobClientBuilder = new SpecializedBlobClientBuilder()
+        .endpoint(azureBlobStorageDestinationConfig.getEndpointUrl())
+        .credential(credential)
+        .containerName(
+            azureBlobStorageDestinationConfig
+                .getContainerName());// Like schema (or even oracle user) in DB
+
+    for (ConfiguredAirbyteStream configuredStream : configuredCatalog.getStreams()) {
+
+      AppendBlobClient appendBlobClient = specializedBlobClientBuilder
+          .blobName(configuredStream.getStream().getName())
+          .buildAppendBlobClient();
+
+      boolean isNewlyCreatedBlob = createContainers(appendBlobClient, configuredStream);
+
+      AzureBlobStorageWriter writer = writerFactory
+          .create(azureBlobStorageDestinationConfig, appendBlobClient, configuredStream,
+              isNewlyCreatedBlob);
+
+      AirbyteStream stream = configuredStream.getStream();
+      AirbyteStreamNameNamespacePair streamNamePair = AirbyteStreamNameNamespacePair
+          .fromAirbyteSteam(stream);
+      streamNameAndNamespaceToWriters.put(streamNamePair, writer);
+    }
+  }
+
+  private boolean createContainers(AppendBlobClient appendBlobClient,
+                                   ConfiguredAirbyteStream configuredStream) {
+    // create container if absent (aka SQl Schema)
+    final BlobContainerClient containerClient = appendBlobClient.getContainerClient();
+    if (!containerClient.exists()) {
+      containerClient.create();
+    }
+    // create a storage container if absent (aka Table is SQL BD)
+    if (configuredStream.getSyncMode().equals(SyncMode.FULL_REFRESH)) {
```

Comment:
Better to write `SyncMode.FULL_REFRESH.equals(configuredStream.getSyncMode())` to avoid possible NPE

Reply:
done, thanks!